Core/clim-core/panes.lisp

```diff
@@ -363,11 +363,46 @@ order to produce a double-click")
   (print-unreadable-object (pane sink :type t :identity t)
     (prin1 (pane-name pane) sink)))
 
+(defgeneric find-concrete-pane-class (pane-realizer pane-type &optional errorp)
+  (:documentation "Resolves abstract pane name to a concrete pane
+class. Methods should specialize on the first argument.")
+  (:method ((realizer t) pane-type &optional (errorp t))
+    ;; Default method tries to resolve the pane name as specified by a
+    ;; convention mentioned in the spec. Function is a little
+    ;; complicated because we preserve old semantics adding rules to
+    ;; the class name resolution. Resolution works as follows:
+    ;;
+    ;; 0. When the pane-type is a class return it as is.
+    ;; 1. Abstract mapping always takes a priority. When it exists we
+    ;;    don't look further.
+    ;; 2. When the symbol is in clim/climi/keyword package:
+    ;;    - look for a class `climi::{SYMBOL-NAME}-pane'
+    ;;    - look for a class `climi::{SYMBOL-NAME}'
+    ;; 3. Otherwise find a class named by the symbol.
+    (when (typep pane-type 'class)
+      (return-from find-concrete-pane-class pane-type))
```

Comment:
This should be a separate method in my opinion.

Reply:
ditto